django/template/base.py

```diff
@@ -393,28 +393,37 @@ def create_token(self, token_string, position, lineno, in_tag):
 
 
 class DebugLexer(Lexer):
+    def _tag_re_split_positions(self):
+        last = 0
+        for match in tag_re.finditer(self.template_string):
+            start, end = match.span()
+            yield (last, start)
+            yield (start, end)
+            last = end
+        yield (last, len(self.template_string))
```

Comment:
We don't need to call `len()`, `slice()` will work the same with `None`.

Reply:
Good point, but do we want the end position to be non-`None` as part of the `position` passed to the token node when debug is true?

Reply:
Agreed, we can't have `None`.

Reply:
This (very) recent PR added a check for 0, and it would be easy to add (or modify) a similar test case for the end-index code path: https://github.com/django/django/pull/14748

Reply:
Good catch :+1: I reverted this change and added a new test case.